### PR TITLE
Feature/opentsdb telnet

### PR DESCRIPTION
**Pull request by phxql**
Hi,
this PR adds the OpenTSDB telnet protocol as an ingester under `/ingest/opentsdb/telnet`.

### CODE REVIEW

File: chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        private final Map<String, String> tags;
+
+        public Metric(String name, Map<String, String> tags) {
+            this.name = name;
+            this.tags = tags;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
```
Comment (sputnikci) [inline chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java]: \[Checkstyle\] ERROR: 'if' construct must use '\{\}'s\.

File: chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+            if (o == null || getClass() != o.getClass()) return false;
```
Comment (sputnikci) [inline chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java]: \[Checkstyle\] ERROR: 'if' construct must use '\{\}'s\.

File: chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                    Objects.equals(tags, metric.tags);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(name, tags);
+        }
+    }
+
+    @Override
```
Comment (sputnikci) [inline chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java]: \[Checkstyle\] ERROR: Fields and methods should be before inner classes\.

File: chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        return metrics.values().stream().map(MetricTimeSeries.Builder::build).collect(Collectors.toList());
+    }
+
+    /**
+     * Extract the metric tags from the parts.
+     *
+     * @param parts Parts.
+     * @return Metric tags.
+     */
+    private Map<String, String> getMetricTags(String[] parts) throws FormatParseException {
```
Comment (sputnikci) [inline chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java]: \[Checkstyle\] ERROR: Fields and methods should be before inner classes\.

File: chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    /**
+     * Extracts the metric timestamp from the parts.
+     *
+     * @param parts Parts.
+     * @return Metric timestamp.
+     * @throws FormatParseException If something went wrong while extracting.
+     */
+    private Instant getMetricTimestamp(String[] parts) throws FormatParseException {
```
Comment (sputnikci) [inline chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java]: \[Checkstyle\] ERROR: Fields and methods should be before inner classes\.

File: chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    /**
+     * Extracts the metric value from the given parts.
+     *
+     * @param parts Parts.
+     * @return Metric value.
+     * @throws FormatParseException If something went wrong while extracting.
+     */
+    private double getMetricValue(String[] parts) throws FormatParseException {
```
Comment (sputnikci) [inline chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java]: \[Checkstyle\] ERROR: Fields and methods should be before inner classes\.

File: chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+        }
+    }
+
+    /**
+     * Extracts the metric name from the given parts.
+     *
+     * @param parts Parts.
+     * @return Metric name.
+     */
+    private String getMetricName(String[] parts) {
```
Comment (sputnikci) [inline chronix-server-ingestion-handler/src/main/java/de/qaware/chronix/solr/ingestion/format/OpenTsdbTelnetFormatParser.java]: \[Checkstyle\] ERROR: Fields and methods should be before inner classes\.

**Comment by FlorianLautenschlager** [comment]
Thx.

**Pull request merged**